rfcs/20220221-3d_model_support.md

```diff
@@ -0,0 +1,181 @@
+| `thumbnail` | `varchar` | Thumbnail URL |
+| `filesize` | `integer` | Size of the file |
+| `license` | `varchar` | License for the model |
+| `license_version` | `varchar` | Version of the model's license |
+| `creator` | `varchar` | Creator ID of the model from the provider |
+| `creator_url` | `varchar` | URL to the creator's page on the provider's site |
+| `title` | `varchar` | Title of the model |
+| `meta_data` | `jsonb` | Uncategorized metadata for the model |
+| `tags` | `jsonb` | Tags that describe the model, for use in searching |
+| `watermarked` | `bool` | Whether the model is watermarked - is this needed for 3D models? |
+| `last_synced_with_source` | `timestamp` | Time the record was last updated from the provider |
+| `removed_from_source` | `bool` | Whether the model has been removed from the upstream source |
+| `category` | `varchar` | Category for the model (is this where we'll make the distinction between 3D printers, game assets, etc.?) |
+| `filetype` | `varchar` | File type of the primary file for the model |
+| `alt_files` | `jsonb` | Additional file types for the model |
+| `vertices` | `integer` | Number of vertices within the model |
+| `textures` | `integer` | Number of [textures](https://en.wikibooks.org/wiki/Blender_3D:_Noob_to_Pro/Materials_and_Textures) the model uses |
+| `materials` | `integer` | Number of [materials](https://en.wikibooks.org/wiki/Blender_3D:_Noob_to_Pro/Materials_and_Textures) the model uses |
+| `uv_mapping` | `bool` | Whether the model contains [UV Mapping](https://en.wikipedia.org/wiki/UV_mapping) |
+| `vertex_colors` | `bool` | Whether the model contains [vertex colors](https://help.sketchfab.com/hc/en-us/articles/210765623-Vertex-Colors) |
+| `pbr` | `bool` | Whether the model supports [physically based rendering](https://en.wikipedia.org/wiki/Physically_based_rendering) |
+| `rigged_geometries` | `bool` | Whether the model supports [rigging for animation](https://dreamfarmstudios.com/blog/what-is-3d-rigging/) |
+| `animations` | `integer` | Number of animations the model has |
+| `morph_geometries` | `integer` | Number of [morph targets for animation](https://en.wikipedia.org/wiki/Morph_target_animation) |
+
+
+# 3D Models in the frontend
+
+## Some potential risks
+
+- 3D models are often heavy files. We should lazy-load and defer loading of these resources as much as possible.
+- We should probably use *thumbnails* rather than actually render 3D models in the search results. We *could* consider something like [SketchFab's search results](https://sketchfab.com/search?q=snacks&type=models), where they render a 3D model on hover of the search results, when on desktop.
+- We'll need to find a standardized 3d model viewer that will work with various formats. We *really, really* don't want to write our own code here. We will also need to decide if we want to use this universally, or use SketchFab's viewer for SketchFab models, which is an excellent viewer.
```

Comment:
If we could flesh out more details about graceful degradation on low power devices that would be great. As mentioned above, 3d models can be heavy assets. We should avoid crashing someones low-power phone by loading too many (or potentially _any_) of them. Unless there are ways to process the assets that could accommodate low memory or small CPU; for example, is it possible to automatically decrease the number of vertices on a model or something like that?